src/api_helpers.jl

```diff
@@ -237,6 +237,16 @@ end
 ### Dataspace Interface
 ###
 
+function h5s_get_regular_hyperslab(space_id)
```

Comment:
Working on https://github.com/JuliaIO/HDF5.jl/pull/749  perhaps we should do the conversions to `Int` in the `api_helpers` interface?  

It's quite messy to have to wrap or truncate constants to `Int` at the highest levels of the API, and makes interacting with the low-level functions more annoying.

Reply:
Looking again basically this is similar to `https://github.com/JuliaIO/HDF5.jl/pull/749` where if you want to practically use the output you have to reverse the dimensions and truncate the data to `Int`, which makes me think we should just do it in the intermediate level API instead so that the display function has to do less wrangling?

Reply:
Perhaps the best option would be to add a higher level wrapper `get_simple_extent_dims` similar to `get_dims` that handles all the data conversion (`get_dims` probably should be rename to `get_simple_extent_dims` ) ?

Reply:
How about making possibly adding a `get_regular_hyperslab` part of #749? That way this PR can just be a simple "add show" and doesn't mess with possibly more holistic API changes.